### ISSUE TITLE
Webview animation is slow on Android

### ISSUE THREAD

**Issue by tokmo** [flutter/flutter#52211]
Hi,

I use webview_flutter plugin for load local html/javascript with js animation.
Animation is really fast on iPhone (6) app or when i open html with chrome on Android but really slow with the Android app (with my galaxy s9).

**Comment by TahaTesser** [comment]
Hi @tokmo 
can you please provide your `flutter doctor -v` and `flutter run --verbose`? 
Also, to better address the issue, would be helpful if you could post a minimal code sample to reproduce the problem
Thank you

**Comment by tokmo** [comment]
```
λ flutter doctor -v
[√] Flutter (Channel stable, v1.12.13+hotfix.8, on Microsoft Windows [version 10.0.18363.693], locale fr-FR)
    • Flutter version 1.12.13+hotfix.8 at C:\flutter
    • Framework revision 0b8abb4724 (4 weeks ago), 2020-02-11 11:44:36 -0800
    • Engine revision e1e6ced81d
    • Dart version 2.7.0


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\Users\capta\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 43.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] VS Code (version 1.42.1)
    • VS Code at C:\Users\capta\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.8.1

[√] Connected device (1 available)
    • SM G960F • 21f09ae4bf0d7ece • android-arm64 • Android 10 (API 29)

• No issues found!
```


The code i use for the webview : 

```
import 'dart:convert';
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:webview_flutter/webview_flutter.dart';
// import 'package:audioplayers/audioplayers.dart';
import 'package:audioplayers/audio_cache.dart';

class JeuLettresImage extends StatefulWidget {
  const JeuLettresImage({Key key, this.categorie, this.lettre});

  final String categorie;
  final String lettre;

  @override
  _JeuLettresImageState createState() => _JeuLettresImageState();
}

class _JeuLettresImageState extends State<JeuLettresImage> {
  WebViewController _controller;
  static AudioCache player = AudioCache();

  @override
  Widget build(BuildContext context) {
    

    return Scaffold(
      resizeToAvoidBottomPadding: false,
      floatingActionButton: Container(
        margin:EdgeInsets.only(top:MediaQuery.of(context).size.width / 100 * 0.42),
        height:MediaQuery.of(context).size.width / 100 * 5.8,
        width:MediaQuery.of(context).size.width / 100 * 5.8,
        // color:Colors.limeAccent,
        child: GestureDetector(
          child:Image.asset('assets/img/sousMenus/backButton.png'),
          onTap: () {
            Navigator.pop(context);
          }
        )
      ),
      // floatingActionButtonLocation: FloatingActionButtonLocation.endTop,
      body: Container(
        
        child: WebView(

          initialUrl: 'about:blank',
          javascriptMode: JavascriptMode.unrestricted,
          
          onWebViewCreated: (WebViewController webViewController) {
            _controller = webViewController;
            _loadHtmlFromAssets();
          },
          onPageFinished: (url) async {
            try {
              var javascript = '''
                    window.alert = function (str){
                      Alert.postMessage(str);
                    }
                  ''';
              await _controller?.evaluateJavascript(javascript);
            } catch (_) {}
            


            // try {
            //   var testAlert = '''
            //         alert('Test alert Message 2');
            //       ''';
            //   await _controller?.evaluateJavascript(testAlert);
            // } catch (_) {}
          },
          javascriptChannels: Set.from(
            [
              JavascriptChannel(
                name: 'Alert',
                onMessageReceived: (JavascriptMessage message) {
                  // alert message = Test alert Message
                  print(message.message);
                  
                  // TODO popup
                  
                },
              ),
              JavascriptChannel(
                name: 'gameEngineStart',
                onMessageReceived: (JavascriptMessage message) {
                  var javascript2 = '''
                    initialisation("''' + widget.lettre + '''", true);
                  ''';
                  _controller?.evaluateJavascript(javascript2);
                }
              ),
              JavascriptChannel(
                name: 'SoundLettreStart',
                onMessageReceived: (JavascriptMessage message) {
                  print(message.message);
                  // String sonLettreChemin = "sons/lettres/", ".mp3",

                  player.play('sons/lettres/' + message.message.toLowerCase() + '.mp3');
                }
              )
            ],
          ),

        ),
      )
    );
  }


  _loadHtmlFromAssets() async {
    String fileText;
    if(widget.categorie == 'avecImage'){
      fileText = await rootBundle.loadString('assets/HTML/jeuLettres/lettreAvecImage.html');
    }
    
    _controller.loadUrl( Uri.dataFromString(
        fileText,
        mimeType: 'text/html',
        encoding: Encoding.getByName('utf-8')
    ).toString());
  }

}

```

GPU acceleration work but i think only 1 process work.

**Comment by TahaTesser** [comment]
Hi @tokmo 
Your code is incomplete, please post minimal code sample 
Thank you

**Comment by tokmo** [comment]
Dart file : 

> import 'dart:convert';
> import 'package:flutter/material.dart';
> import 'package:flutter/services.dart';
> import 'package:webview_flutter/webview_flutter.dart';
> // import 'package:audioplayers/audioplayers.dart';
> import 'package:audioplayers/audio_cache.dart';
> 
> class JeuLettresImage extends StatefulWidget {
>   const JeuLettresImage({Key key, this.categorie, this.lettre});
> 
>   final String categorie;
>   final String lettre;
> 
>   @override
>   _JeuLettresImageState createState() => _JeuLettresImageState();
> }
> 
> class _JeuLettresImageState extends State<JeuLettresImage> {
>   WebViewController _controller;
>   static AudioCache player = AudioCache();
> 
>   @override
>   Widget build(BuildContext context) {
>     
> 
>     return Scaffold(
>       resizeToAvoidBottomPadding: false,
>       floatingActionButton: Container(
>         margin:EdgeInsets.only(top:MediaQuery.of(context).size.width / 100 * 0.42),
>         height:MediaQuery.of(context).size.width / 100 * 5.8,
>         width:MediaQuery.of(context).size.width / 100 * 5.8,
>         // color:Colors.limeAccent,
>         child: GestureDetector(
>           child:Image.asset('assets/img/sousMenus/backButton.png'),
>           onTap: () {
>             Navigator.pop(context);
>           }
>         )
>       ),
>       // floatingActionButtonLocation: FloatingActionButtonLocation.endTop,
>       body: Container(
>         
>         child: WebView(
> 
>           initialUrl: 'about:blank',
>           javascriptMode: JavascriptMode.unrestricted,
>           
>           onWebViewCreated: (WebViewController webViewController) {
>             _controller = webViewController;
>             _loadHtmlFromAssets();
>           },
>           javascriptChannels: Set.from(
>             [
>             
>               JavascriptChannel(
>                 name: 'gameEngineStart',
>                 onMessageReceived: (JavascriptMessage message) {
>                   var javascript2 = '''
>                     initialisation("''' + widget.lettre + '''", true);
>                   ''';
>                   _controller?.evaluateJavascript(javascript2);
>                 }
>               ),
>               
>             ],
>           ),
> 
>         ),
>       )
>     );
>   }
> 
> 
>   _loadHtmlFromAssets() async {
>     String fileText;
>     if(widget.categorie == 'avecImage'){
>       fileText = await rootBundle.loadString('assets/HTML/jeuLettres/lettreAvecImage.html');
>     }
>     
>     _controller.loadUrl( Uri.dataFromString(
>         fileText,
>         mimeType: 'text/html',
>         encoding: Encoding.getByName('utf-8')
>     ).toString());
>   }
> }

In the HTML i have SVG inline.
In my javascript i make scale (or scale 3D) translate (or translate3d) for anime the SVG inline inside the HTML

**Comment by shinsenter** [comment]
I also have a similar laggy problem with webview on Android (tested with xperia z1 and galaxy note 10) when scrolling web pages. My website has no animation, it's just a plain html page. It runs perfectly on iOS or Google Chrome.

**Comment by deandreamatias** [comment]
Maybe related with #29892

**Comment by darshankawar** [comment]
Hi @tokmo,
Can you please confirm if you still experience this issue on latest Flutter version and latest `webview_flutter` plugin ?
If so, it'll be good to share a minimal reproducible code sample that'll help to replicate issue.

**Comment by TahaTesser** [comment]
Without additional information we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

**Comment by blasten** [comment]
This has been fixed in Flutter 1.22/ WebView 1.0 - use hybrid composition in the Flutter WebView plugin.

https://github.com/flutter/plugins/tree/master/packages/webview_flutter#using-hybrid-composition
